### PR TITLE
ENT-3479 | Added save and add more functionality in enterprise coupon…

### PR DESCRIPTION
**Pull request by hasnain-naveed**
… creation page.

### CODE REVIEW

File: ecommerce/static/js/views/enterprise_coupon_form_view.js [hunk 1]
```diff
@@ -267,7 +272,27 @@ define([
                     this.setLimitToElement(this.$('[name=contract_discount_value]'), 100, 0);
                 }
                 this._super();
+                if (this.editing) {
+                    this.$('.submit-add-more').hide();
+                } else {
+                    this.$('.submit-add-more').html(gettext('Create and Add More'));
```
Comment (sameenfatima78) [inline ecommerce/static/js/views/enterprise_coupon_form_view.js:279]: Nit: I think 'Create and add another' would make more sense.

**Review by sameenfatima78 — CHANGES_REQUESTED** [review]
Shouldn't we add test cases to ensure that this new functionality is working as expected?

**Comment by hasnain-naveed** [comment]
@sameenfatima78 yes, you are right but there are no test cases for this file enterprise_coupon_form_view, you can check the code base.

**Comment by sameenfatima78** [comment]
@hasnain-naveed I think they are in this file: https://github.com/edx/ecommerce/blob/master/ecommerce/static/js/test/specs/views/enterprise_coupon_create_edit_view_spec.js

**Comment by hasnain-naveed** [comment]
> @hasnain-naveed I think they are in this file: https://github.com/edx/ecommerce/blob/master/ecommerce/static/js/test/specs/views/enterprise_coupon_create_edit_view_spec.js

These are the model save test cases. I just added a new button I can add a test case that will be ensuring that the button is visible in case of creation and disable in case of the edit.

File: ecommerce/static/js/views/enterprise_coupon_form_view.js (hunk 1 above)
Comment (hasnain-naveed) [reply ecommerce/static/js/views/enterprise_coupon_form_view.js:279]: It is in A/C/.

**Review by sameenfatima78 — APPROVED** [review]

**Review by saleem-latif — APPROVED** [review]

**Pull request merged**